### PR TITLE
Joe unit methods 2

### PR DESCRIPTION
**Pull request by JosiahDeYong**

### CODE REVIEW

File: TAScheduler/views_methods.py [hunk 1]
```diff
@@ -404,6 +405,37 @@ def courseTAAsmgt(self, active_ta, active_course):  # new remove sectionTAAsmgt
...
+    def sectionTAAsmgt(self, active_ta, active_section):
+        if not isinstance(active_ta, TA):
+            raise TypeError("Input passed to admin.sectionTAAsmgt is not TA")
+        if not isinstance(active_section, LabObj) and not isinstance(active_section, LectureObj):
+            raise TypeError("Input passed to admin.sectionTAAsmgt is not Lab or Lecture")
+        active_section.addTA(active_ta)
+
+    def getAllCrseAsgmts(self):
+        outputdict = {}
+        if InstructorToCourse.objects.all().count() + TAToCourse.objects.all().count() == 0:
```
Comment (AlecBrookens) [inline TAScheduler/views_methods.py:417]: you could use .exists() rather than count and compare the number

File: TAScheduler/views_methods.py [hunk 2]
```diff
@@ -404,6 +405,37 @@ def courseTAAsmgt(self, active_ta, active_course):  # new remove sectionTAAsmgt
...
+            raise RuntimeError("No course links exist")
+        for i in InstructorToCourse.objects.all():
+            outputdict[i.course.course_id] = i.instructor.user.email_address
+        for i in TAToCourse.objects.all():
+            if i.course.course_id in outputdict:
+                outputdict[i.course.course_id] = (outputdict[i.course.course_id], i.ta.user.email_address)
+            else:
+                outputdict[i.course.course_id] = i.ta
```
Comment (AlecBrookens) [inline TAScheduler/views_methods.py:425]: can replace with 
outputdict.setdefault(i.course.course_id, []).append(i.ta.user.email_address) 
for more consistency

File: TAScheduler/views_methods.py [hunk 3]
```diff
@@ -394,7 +394,8 @@ def courseInstrAsmgt(self, active_instr, active_course):  # new inputs
     def courseTAAsmgt(self, active_ta, active_course):  # new remove sectionTAAsmgt
         if not isinstance(active_ta, TAObj):
             raise TypeError("Input passed is not an object of ta obj")
-        elif not User.objects.filter(email_address=active_ta.getUsername()).exists():
+        elif not User.objects.filter(email_address=active_ta.getUsername()).exists():  # I believe this is a redundant
```
Comment (Kiran-Ee) [inline TAScheduler/views_methods.py:397]: Yeah i think the db takes care of this already

File: TAScheduler/views_methods.py [hunk 4]
```diff
@@ -404,6 +405,37 @@ def courseTAAsmgt(self, active_ta, active_course):  # new remove sectionTAAsmgt
             raise RuntimeError("Instructor is already assigned to max number of course permitted")
         TAToCourse.objects.create(ta=active_ta.database, course=active_course.database)
 
+    def sectionTAAsmgt(self, active_ta, active_section):
```
Comment (Kiran-Ee) [inline TAScheduler/views_methods.py:408]: I think we usually want our assignment methods to take in "objs" not DBObjs - I see you're comparing active_ta to "TA" (not TAObj) and then you're checking active_section to LabObj

I would recommend comparing them to the class objects

File: TAScheduler/views_methods.py [hunk 5]
```diff
@@ -404,6 +405,37 @@ def courseTAAsmgt(self, active_ta, active_course):  # new remove sectionTAAsmgt
...
+        for i in InstructorToCourse.objects.all():
+            outputdict[i.course.course_id] = i.instructor.user.email_address
+        for i in TAToCourse.objects.all():
+            if i.course.course_id in outputdict:
+                outputdict[i.course.course_id] = (outputdict[i.course.course_id], i.ta.user.email_address)
+            else:
+                outputdict[i.course.course_id] = i.ta
+        return outputdict
+
+    def courseUserAsgmt(self, active_user, active_course):
```
Comment (Kiran-Ee) [inline TAScheduler/views_methods.py:428]: same thing here too ^

**Review by Kiran-Ee — COMMENTED** [review]
Adjust assign methods + recommendations

File: TAScheduler/views_methods.py (hunk 4 above)
Comment (JosiahDeYong) [reply TAScheduler/views_methods.py:408]: The way I made LabObj and LecObj work is they take the database, but I can change them to take TAObj if you want?

File: TAScheduler/views_methods.py (hunk 5 above)
Comment (JosiahDeYong) [reply TAScheduler/views_methods.py:428]: Ok I changed it just when this gets merged with master I've gotta fix the unit tests for this

**Pull request closed without merging**